1.train_models/log_reg_plate_3_3p_cp_renorm_fs_data/nbconverted/train_model.py

```diff
@@ -0,0 +1,247 @@
+
+
+# ## Import Utils
+
+# In[ ]:
+
+
+sys.path.append(f"{root_dir}/utils")
+import preprocess_utils as ppu
+
+
+# # Seed and Generator for Reproducibility
+
+# In[ ]:
+
+
+rnd_val = 0  # Random value for all seeds
+rng = np.random.default_rng(seed=rnd_val)  # random number generator
+
+
+# # Converting parquet to pandas dataframe
+
+# ## Feature selected plate data
+
+# In[ ]:
+
+
+filename3 = "Plate_3_sc_norm_fs.parquet"
+filename3p = "Plate_3_prime_sc_norm_fs.parquet"
+plate_path = Path(
+    f"{root_dir}/nf1_painting_repo/3.processing_features/data/feature_selected_data"
+)
+
+path3 = plate_path / filename3
+
+path3p = plate_path / filename3p
+
+# Creates an object for accessing the plate 3 normalized data using the path of the data
+po3 = ppu.Preprocess_data(path=path3)
+
+# Creates an object for accessing the plate 3 prime normalized data using the path of the data
+po3p = ppu.Preprocess_data(path=path3p)
+
+plate3df = po3.df  # Returns the dataframe generated by the csv
+plate3pdf = po3p.df  # Returns the dataframe generated by the csv
+
+common_columns = list(plate3df.columns.intersection(plate3pdf.columns))
+
+
+# ## Annotated plate data
+
+# In[ ]:
+
+
+filename3 = "Plate_3_sc.parquet"
+filename3p = "Plate_3_prime_sc.parquet"
+plate_path = Path(
+    f"{root_dir}/nf1_painting_repo/3.processing_features/data/annotated_data"
+)
+
+path3 = plate_path / filename3
+
+path3p = plate_path / filename3p
+
+# Creates an object for accessing the plate 3 normalized data using the path of the data
+po3 = ppu.Preprocess_data(path=path3)
+
+# Creates an object for accessing the plate 3 prime normalized data using the path of the data
+po3p = ppu.Preprocess_data(path=path3p)
+
+plate3df = po3.df  # Returns the dataframe generated by the csv
+plate3pdf = po3p.df  # Returns the dataframe generated by the csv
+
+
+# # Preprocess Data
+
+# ## Use only common columns
+
+# In[ ]:
+
+
+# Set plate column:
+plate3df["Metadata_plate"] = "3"
+plate3pdf["Metadata_plate"] = "3p"
+
+plate3df = plate3df.loc[:, common_columns]
+plate3pdf = plate3pdf.loc[:, common_columns]
+
+# Combine the plate dataframes:
+platedf = pd.concat([plate3df, plate3pdf], axis="rows")
+
+
+# ## Normalize Data
+
+# In[ ]:
+
+
+# Get all columns that aren't metadata
+columns_to_normalize = [col for col in platedf.columns if "Metadata" not in col]
+
+# Normalize the columns
+mms = MinMaxScaler()
+normdf = pd.DataFrame(
+    mms.fit_transform(platedf[columns_to_normalize]), columns=columns_to_normalize
+)
+
+# Apply the transformation to the dataframe
+platedf[columns_to_normalize] = normdf
+
+
+# ## Create Classes
+
+# In[ ]:
+
+
+target_column = "Metadata_genotype"
+stratify_column = "Metadata_Well"
+
+# These represent the fractions of the entire dataset
+train_val_frac = 0.85
+test_frac = 1 - train_val_frac
+val_frac = 0.15
+
+
+# ## Down-sample and stratify by well
+
+# In[ ]:
+
+
+smallest_gene = platedf[target_column].value_counts().min()
+platedata = pd.DataFrame()
+
```

Comment:
consider adding a comment of what is occurring within the for loop

Reply:
Added!